faiss/gpu/GpuIndex.cu

```diff
@@ -42,6 +42,17 @@ constexpr idx_t kAddVecSize = (idx_t)512 * 1024;
 // FIXME: parameterize based on algorithm need
 constexpr idx_t kSearchVecSize = (idx_t)32 * 1024;
 
+bool should_use_raft(GpuIndexConfig config_) {
+#ifdef __CUDA_ARCH__
```

Comment:
@tfeher Can I define a new host macro that can be used in host functions? What I am trying to say is that while __CUDA_ARCH__ is not used directly in host functions, there are examples of it being added at the start of header files. For example
https://github.com/facebookresearch/faiss/blob/ebb5f8401bdff6b5802738bb93385963e0f73c33/faiss/gpu/impl/PQCodeLoad.cuh#L15.
If this works, can we have a bool macro like CUDA_ARCH_GREATER_THAN_700 that can be used in host functions?

Reply:
The linked code is incorrect. In host code `__CUDA_ARCH__ ` is undefined. The preprocessor replaces undefined values with 0 before [evaluating the expression](https://stackoverflow.com/a/5085425), therefore it does not throw error, but it does not have the desired behavior. 

I think what we need here is not a preprocessor directive, just a runtime bool flag. We could just query the device properties and decide accordingly

```c++
  if (resource::get_device_properties(handle).major < 7) return false;
```

I am using here the raft `handle `because it caches the device properties.

Reply:
Done.